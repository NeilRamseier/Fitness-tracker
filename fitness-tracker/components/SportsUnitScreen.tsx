import { View, Text } from "react-native";
import { Menu, useTheme, RadioButton } from "react-native-paper";
import * as React from 'react';
import { TextInput, Button } from 'react-native-paper';

export default function SportsUnitScreen() {

  const theme = useTheme();
  const [activityValue, setActivityValue] = React.useState("");
  const [distance, setDistance] = React.useState("");
  const [time, setTime] = React.useState("");
  const [inputError, setInputError] = React.useState(true);


  const handleSave = () => {
    const distanceNumber = parseFloat(distance);
    const timeNumber = parseFloat(time)

    if (!isNaN(distanceNumber) && !isNaN(timeNumber) && (activityValue != "")) {
      setInputError(false);
    } else{
      setInputError(true);
    }
  }

  return (
    <View style={{ backgroundColor: theme.colors.primary ,flex: 1, justifyContent: "flex-start", alignItems: "center", }}>
      <Text style={{ color: theme.colors.secondary, margin: 15 }} >Man kann erst speichern wenn in jedem Feld ein Korrekter Wert angegeben wurde. </Text>
      <Text style={{ color: theme.colors.secondary, fontSize: 25, marginTop: 15 }}>Aktivit√§t</Text>
      <View style={{ backgroundColor: theme.colors.secondary, justifyContent: "flex-start", height: 200, marginTop: 15 }}>
      <RadioButton.Group onValueChange={activityValue => setActivityValue(activityValue)} value={activityValue} >
        <RadioButton.Item label="Joggen" value="Joggen" />
        <RadioButton.Item label="Fahrrad fahren" value="Fahrrad fahren" />
        <RadioButton.Item label="Schwimmen" value="Schwimmen" />
       <RadioButton.Item label="Gehen" value="Gehen" />
      </RadioButton.Group>
      </View>
      <Text style={{ color: theme.colors.secondary, fontSize: 25, marginTop: 30 }}>Strecke</Text>
      <TextInput
      style = {{
        marginTop: 15,
        width: 182,
        height: 50
      }}
      placeholder="Km"
      value={distance}
      onChangeText={distance => setDistance(distance)}
    />
          <Text style={{ color: theme.colors.secondary, fontSize: 25, marginTop: 30 }}>Dauer</Text>
      <TextInput
      style = {{
        marginTop: 15,
        width: 182,
        height: 50
      }}
      placeholder="min"
      value={time}
      onChangeText={time => setTime(time)}
    />
    {/* Der Button muss onPress() noch modifiziert werden. */}
    <Button mode="contained" onPress={() => console.log(handleSave)} style= {{
      justifyContent: "center",
      borderWidth: 1,
      borderColor: theme.colors.secondary,
      width: 182,
      height: 50,
      marginTop: 30
    }}>
    Speichern
  </Button>
    </View>
  );
  }