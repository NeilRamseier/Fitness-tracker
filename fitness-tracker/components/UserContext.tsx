// UserContext.tsx

import React, { createContext, useContext, useState, ReactNode } from 'react';

interface User {
  id?: number;
  first_name?: string;
  last_name?: string;
  weight?: number;
  height?: number;
  age?: number;
  gender?: string;
  basal_metabolic_rate?: number;
  IP?: string;
  creation_date?: string;
}

interface UserContextType {
  user: User;
  setUser: React.Dispatch<React.SetStateAction<User>>;
  calories: number | null;
  setCalories: React.Dispatch<React.SetStateAction<number | null>>;
}

const UserContext = createContext<UserContextType | undefined>(undefined);

export const UserProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
  const [user, setUser] = useState<User>({});
  const [calories, setCalories] = useState<number | null>(null);

  return (
    <UserContext.Provider value={{ user, setUser, calories, setCalories }}>
      {children}
    </UserContext.Provider>
  );
};

export const useUser = (): UserContextType => {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error('useUser must be used within a UserProvider');
  }
  return context;
};
