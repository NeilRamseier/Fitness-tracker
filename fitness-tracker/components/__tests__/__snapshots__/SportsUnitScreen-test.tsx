// SportsUnitScreen.test.tsx
import React from 'react';
import renderer from 'react-test-renderer';
import SportsUnitScreen from '../../SportsUnitScreen';
import { useUser } from '../../UserContext';

jest.useFakeTimers();

jest.mock('../../UserContext', () => ({
  useUser: jest.fn(() => ({
    user: { weight: 70 },
    setCalories: jest.fn(),
  })),
}));

describe('<SportsUnitScreen />', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<SportsUnitScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
