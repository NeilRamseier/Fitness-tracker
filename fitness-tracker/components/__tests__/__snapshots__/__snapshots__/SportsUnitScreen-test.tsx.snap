// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SportsUnitScreen /> should match the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(103, 80, 164, 1)",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <Text
    style={
      {
        "color": "rgba(98, 91, 113, 1)",
        "margin": 15,
      }
    }
  >
    Man kann erst speichern, wenn in jedem Feld ein korrekter Wert angegeben wurde. 
  </Text>
  <Text
    style={
      {
        "color": "rgba(98, 91, 113, 1)",
        "fontSize": 25,
        "marginTop": 15,
      }
    }
  >
    Aktivit√§t
  </Text>
  <View
    style={
      {
        "backgroundColor": "rgba(98, 91, 113, 1)",
        "height": 200,
        "justifyContent": "flex-start",
        "marginTop": 15,
      }
    }
  >
    <View
      accessibilityRole="radiogroup"
    >
      <View
        accessibilityLabel="Joggen"
        accessibilityRole="radio"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            false,
            undefined,
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.15,
                    "lineHeight": 24,
                  },
                  [
                    {
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                    false,
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "textAlign": "left",
                    },
                    undefined,
                  ],
                ],
              ]
            }
          >
            Joggen
          </Text>
          <View
            accessibilityLiveRegion="polite"
            accessibilityRole="radio"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 18,
                  "padding": 6,
                },
              ]
            }
          >
            <View
              style={
                {
                  "opacity": 0,
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityLabel="Fahrrad fahren"
        accessibilityRole="radio"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            false,
            undefined,
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.15,
                    "lineHeight": 24,
                  },
                  [
                    {
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                    false,
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "textAlign": "left",
                    },
                    undefined,
                  ],
                ],
              ]
            }
          >
            Fahrrad fahren
          </Text>
          <View
            accessibilityLiveRegion="polite"
            accessibilityRole="radio"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 18,
                  "padding": 6,
                },
              ]
            }
          >
            <View
              style={
                {
                  "opacity": 0,
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityLabel="Schwimmen"
        accessibilityRole="radio"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            false,
            undefined,
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.15,
                    "lineHeight": 24,
                  },
                  [
                    {
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                    false,
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "textAlign": "left",
                    },
                    undefined,
                  ],
                ],
              ]
            }
          >
            Schwimmen
          </Text>
          <View
            accessibilityLiveRegion="polite"
            accessibilityRole="radio"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 18,
                  "padding": 6,
                },
              ]
            }
          >
            <View
              style={
                {
                  "opacity": 0,
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityLabel="Gehen"
        accessibilityRole="radio"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            false,
            undefined,
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.15,
                    "lineHeight": 24,
                  },
                  [
                    {
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                    false,
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "textAlign": "left",
                    },
                    undefined,
                  ],
                ],
              ]
            }
          >
            Gehen
          </Text>
          <View
            accessibilityLiveRegion="polite"
            accessibilityRole="radio"
            accessibilityState={
              {
                "busy": undefined,
                "checked": false,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 18,
                  "padding": 6,
                },
              ]
            }
          >
            <View
              style={
                {
                  "opacity": 0,
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <Text
    style={
      {
        "color": "rgba(98, 91, 113, 1)",
        "fontSize": 25,
        "marginTop": 30,
      }
    }
  >
    Strecke
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(231, 224, 236, 1)",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        {
          "marginTop": 15,
          "width": 182,
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(73, 69, 79, 1)",
          "bottom": 0,
          "height": 1,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": [
            {
              "scaleY": 0.5,
            },
          ],
          "zIndex": 1,
        }
      }
      testID="text-input-underline"
    />
    <View
      onLayout={[Function]}
      style={
        [
          {
            "flexGrow": 1,
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          {
            "minHeight": 50,
          },
        ]
      }
    >
      <TextInput
        cursorColor="rgba(103, 80, 164, 1)"
        editable={true}
        maxFontSizeMultiplier={1.5}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Km"
        placeholderTextColor="rgba(73, 69, 79, 1)"
        selectionColor="rgba(103, 80, 164, 1)"
        style={
          [
            {
              "flexGrow": 1,
              "margin": 0,
            },
            {},
            {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "letterSpacing": 0.15,
              "lineHeight": undefined,
              "minWidth": 65,
              "paddingLeft": 16,
              "paddingRight": 16,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            [
              {},
            ],
            undefined,
          ]
        }
        testID="text-input-flat"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <Text
    style={
      {
        "color": "rgba(98, 91, 113, 1)",
        "fontSize": 25,
        "marginTop": 30,
      }
    }
  >
    Dauer
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(231, 224, 236, 1)",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        {
          "marginTop": 15,
          "width": 182,
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(73, 69, 79, 1)",
          "bottom": 0,
          "height": 1,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": [
            {
              "scaleY": 0.5,
            },
          ],
          "zIndex": 1,
        }
      }
      testID="text-input-underline"
    />
    <View
      onLayout={[Function]}
      style={
        [
          {
            "flexGrow": 1,
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          {
            "minHeight": 50,
          },
        ]
      }
    >
      <TextInput
        cursorColor="rgba(103, 80, 164, 1)"
        editable={true}
        maxFontSizeMultiplier={1.5}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="min"
        placeholderTextColor="rgba(73, 69, 79, 1)"
        selectionColor="rgba(103, 80, 164, 1)"
        style={
          [
            {
              "flexGrow": 1,
              "margin": 0,
            },
            {},
            {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "letterSpacing": 0.15,
              "lineHeight": undefined,
              "minWidth": 65,
              "paddingLeft": 16,
              "paddingRight": 16,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            [
              {},
            ],
            undefined,
          ]
        }
        testID="text-input-flat"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(103, 80, 164, 1)",
        "borderRadius": 20,
        "height": 50,
        "marginTop": 30,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "width": 182,
      }
    }
    testID="button-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(103, 80, 164, 1)",
          "borderColor": "rgba(98, 91, 113, 1)",
          "borderRadius": 20,
          "borderStyle": "solid",
          "borderWidth": 1,
          "flex": 1,
          "justifyContent": "center",
          "minWidth": 64,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 20,
            },
          ]
        }
        testID="button"
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  [
                    {
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    false,
                    {
                      "marginHorizontal": 24,
                      "marginVertical": 10,
                    },
                    undefined,
                    false,
                    {
                      "color": "rgba(255, 255, 255, 1)",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ],
              ]
            }
            testID="button-text"
          >
            Speichern
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
